File: core-examples/src/main/java/io/vertx/example/core/http/sendfile/SendFile.java
package io.vertx.example.core.http.sendfile;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Launcher;

/*
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class SendFile extends AbstractVerticle {

  public static void main(String[] args) {
    Launcher.executeCommand("run", SendFile.class.getName());
  }

  @Override
  public void start() throws Exception {

    // In reality, it's highly recommend you use Vert.x-Web for applications like this.

    vertx.createHttpServer().requestHandler(req -> {
      String filename = null;
      if (req.path().equals("/")) {
        filename = "index.html";
      } else if (req.path().equals("/page1.html")) {
        filename = "page1.html";
      } else if (req.path().equals("/page2.html")) {
        filename = "page2.html";
      } else {
        req.response().setStatusCode(404).end();
      }
      if (filename != null) {
        req.response().sendFile("io/vertx/example/core/http/sendfile/" + filename);
      }
    }).listen(8080);
  }
}

File: core-examples/src/main/kotlin/io/vertx/example/core/http/sendfile/SendFile.kt
package io.vertx.example.core.http.sendfile

import io.vertx.kotlin.common.json.*

class SendFile : io.vertx.core.AbstractVerticle()  {
  override fun start() {

    // In reality it's highly recommend you use Vert.x-Web for applications like this.

    vertx.createHttpServer().requestHandler({ req ->
      var filename: String? = null
      if (req.path() == "/") {
        filename = "index.html"
      } else if (req.path() == "/page1.html") {
        filename = "page1.html"
      } else if (req.path() == "/page2.html") {
        filename = "page2.html"
      } else {
        req.response().setStatusCode(404).end()
      }
      if (filename != null) {
        req.response().sendFile(filename)
      }
    }).listen(8080)
  }
}
